enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,543 @@
+
+### Test Plan
+
+We will add a new test to the OpenShift Machine API E2E test suite that runs against both the Machine API Operator and
+Machine API Provider AWS repositories.
+
+As there should be little to no difference with regards to the different placement groups from the Machine API
+perspective, we will only test one of the placement group types.
+
+The test will:
+- Create a new MachineSet with 2 replicas and a Partition Placement Group
+- Ensure that the Machine instances are created successfully
+- Remove the first Machine by scaling the MachineSet to 1 replica and wait for its removal
+- Check there are no issue with the first Machine removal
+- Check that the second Machine is still Running as expected
+- Remove the MachineSet and wait for it the remaining Machine to be removed
+- Check that the Placement group has been removed along with the Second Machine
+
+This will ensure that we have the appropriate permissions to perform the required actions for placement groups as well
+as that our removal logic does not cause issues.
+
+### Graduation Criteria
+
+As with other features of this nature, this will become GA from the initial release. There will be no graduation
+criteria.
+
+#### Dev Preview -> Tech Preview
+
+N/A as per the above.
+
+#### Tech Preview -> GA
+
+N/A as per the above.
+
+#### Removing a deprecated feature
+
+This enhancement does not describe the removal of a deprecated feature.
+
+### Upgrade / Downgrade Strategy
+
+As this is a new feature, we do not expect any issues during upgrades.
+There will be additional permissions required by the Machine API provider AWS, but existing Cloud Credential Operator
+upgrade gating should handle this in manual mode where it may cause issues.
+
+On downgrade, if a user had created an instance in a placement group, the instance will remain functional. There should
+be no immediate impact on the running cluster.
+The only issue that may arise is that the Machine API provider will no longer know about the placement group that it
+created.
+In this case, when the last Machine is removed, we will orphan the placement group, it is expected that the end user
+will be responsible for removing this in this scenario.
+
+### Version Skew Strategy
+
+There are no cross component dependencies and as such, we do not expect any cross version issues.
+
+### Operational Aspects of API Extensions
+
+We do not expect any need for operation of the API extensions introduced in this enhancement.
+
+#### Failure Modes
+
+The only known failure mode is when limits are reached for the number of instances within the placement groups. This is
+already discussed in the Risks and Mitigations section.
+
+This failure will prevent adding additional compute capacity to the cluster and in turn, will result in potentially
+pending workloads.
+As this feature is only supported on worker nodes, we do not expect any detrimental impact on the control plane.
```

Comment:
It may not be supported, but is it possible for users to configure this on the control-plane Machines, or will we block this somehow?

Reply:
Not supported for the moment, this is blocked purely on the fact that we aren't going to implement this in the installer. I've added some notes about control plane machines to the future implementation and drawbacks sections